File: app/backend/java/payment-service/src/main/java/com/srbms/paymentservice/dto/RefundRequestDTO.java
package com.srbms.paymentservice.dto;

import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;

public class RefundRequestDTO {

    @NotNull(message = "Payment ID is required")
    private Integer paymentId;

    @DecimalMin(value = "0.01", message = "Refund amount must be greater than zero")
    private BigDecimal refundAmount; // Optional for partial refund, null for full refund

    @NotNull(message = "Reason for refund is required")
    private String reason;

    public RefundRequestDTO() {}

    public Integer getPaymentId() {
        return paymentId;
    }

    public void setPaymentId(Integer paymentId) {
        this.paymentId = paymentId;
    }

    public BigDecimal getRefundAmount() {
        return refundAmount;
    }

    public void setRefundAmount(BigDecimal refundAmount) {
        this.refundAmount = refundAmount;
    }

    public String getReason() {
        return reason;
    }

    public void setReason(String reason) {
        this.reason = reason;
    }
}

File: app/backend/.net/SmartRide.PaymentService/Models/DTOs/RefundRequestDTO.cs
using System.ComponentModel.DataAnnotations;

namespace SmartRide.PaymentService.Models.DTOs;

public class RefundRequestDTO
{
    [Required(ErrorMessage = "Payment ID is required")]
    public int PaymentId { get; set; }

    [Range(typeof(decimal), "0.01", "79228162514264337593543950335", ErrorMessage = "Refund amount must be greater than zero")]
    public decimal? RefundAmount { get; set; }

    [Required(ErrorMessage = "Reason for refund is required")]
    public string Reason { get; set; } = string.Empty;
}
